read.go

```diff
@@ -491,20 +491,25 @@ func readElement(r dicomio.Reader, d *Dataset, fc chan<- *frame.Frame) (*Element
 		return nil, err
 	}
 	// log.Println("readElement: readTag: ", t)
+	readImplicit := r.IsImplicit()
+	if *t == tag.Item {
+		// Always read implicit for item elements
+		readImplicit = true
+	}
```

Comment:
@suyashkumar This should almost solve the issue, but I have one concern; won't the implicit VR setting result in a VR of "NA" getting returned by the readVR function for all tag.Items? This means in the readVL function the code in the first switch statement will execute, and this code calls r.skip(2) because of 2 reserved bytes. I think in the case of an item there are no reserved bytes and the immediate next 4 bytes is the VL as a UINT32. Let me know your thoughts

Reply:
No, I don't think this will happen. If you look at line 51, you'll see that if `isImplicit = true`, then all we do in readVL is read and return a UInt32 (in other words, we do not progress to the lines you are talking about unless we're in an explicit transfer syntax).

https://github.com/suyashkumar/dicom/blob/6fd13fe51495259289c6c8dba3779ee4d3a25d37/read.go#L49-L52

Let me know if I missed something!

Reply:
Ah yes, thanks for pointing that out!